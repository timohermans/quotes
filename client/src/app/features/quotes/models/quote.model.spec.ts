import { Quote } from './quote.model';

describe('QuoteModel', () => {
  it('maps an API resource to a quote object', () => {
    const json = {
      author: 'C. A. R. Hoare',
      quote:
        'There are two ways of constructing a software design; one way is to make it so simple that there are ' +
        'obviously no deficiencies, and the other way is to make it so complicated that there are no obvious ' +
        'deficiencies. The first method is far more difficult.',
    };

    const quote = Quote.fromResource(json);

    expect(quote).toMatchSnapshot();
  });

  it('maps a resource quote with votes to a Quote', () => {
    const json = {
      author: 'C. A. R. Hoare',
      quote:
        'There are two ways of constructing a software design; one way is to make it so simple that there are ' +
        'obviously no deficiencies, and the other way is to make it so complicated that there are no obvious ' +
        'deficiencies. The first method is far more difficult.',
      ratingAverage: 4.75,
      amountOfVotes: 4,
    };

    const quote = Quote.fromResource(json);

    expect(quote).toMatchSnapshot();
  });
});
