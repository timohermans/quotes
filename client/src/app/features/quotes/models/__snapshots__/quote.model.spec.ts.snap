// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteModel maps a resource quote with votes to a Quote 1`] = `
Quote {
  "amountOfVotes": 4,
  "author": "C. A. R. Hoare",
  "quote": "There are two ways of constructing a software design; one way is to make it so simple that there are obviously no deficiencies, and the other way is to make it so complicated that there are no obvious deficiencies. The first method is far more difficult.",
  "ratingAverage": 4.75,
}
`;

exports[`QuoteModel maps an API resource to a quote object 1`] = `
Quote {
  "amountOfVotes": undefined,
  "author": "C. A. R. Hoare",
  "quote": "There are two ways of constructing a software design; one way is to make it so simple that there are obviously no deficiencies, and the other way is to make it so complicated that there are no obvious deficiencies. The first method is far more difficult.",
  "ratingAverage": undefined,
}
`;
